File: harness/elixir/microformats2/lib/mf2.ex
defmodule MF2 do
  def main(args) do
    [filename, base, iterations] = args
    contents = File.read!(filename)

    if String.to_integer(iterations) == 0 do
      IO.puts Poison.encode!(Microformats2.parse(contents, base))
    else
      Enum.each(1..String.to_integer(iterations), fn(_) ->
        start = System.monotonic_time(:microsecond)
        _ = Microformats2.parse(contents, base)
        duration = System.monotonic_time(:microsecond) - start
        IO.puts "#{duration}"
      end)
    end
  end
end

File: benches/ruby/microformats-ruby/mf2.rb
require 'microformats'

file = File.open(ARGV[0], "r")
contents = file.read

idx = 0
iterations = ARGV[2].to_i
until idx >= iterations do
	start = Time.now
    Microformats.parse(contents, base: ARGV[1])
	puts ((Time.now - start) * 1000000).truncate
	idx += 1
end
